tools/cmake/utility.cmake

```diff
@@ -1,5 +1,11 @@
 include(CheckSymbolExists)
 
+# This function can be called to define a set of required and optional macros for a build target.
+#
+# If a CMake variable is defined with the same name as a required macro, it will be
```

Comment:
(Unrelated to the changes in this PR) The `CMAKE_REQUIRED_DEFINITIONS` list in this utility function seems unused. Can that be removed?

Reply:
It is actually used by the `check_symbol_exists` function. Although this is not immediately obviously and I must admit that I forgot about it too, so I've left a comment about that.